packages/explorer/src/queries/sqlite-indexer/useDeployedTablesQuery.ts

```diff
@@ -0,0 +1,44 @@
+import { useParams } from "next/navigation";
+import { Hex } from "viem";
+import { useQuery } from "@tanstack/react-query";
+import { decodeTable } from "../../app/(explorer)/[chainName]/worlds/[worldAddress]/explore/utils/decodeTable";
+import { internalMUDNamespaces } from "../../common";
+
+type ApiResponse = {
+  table: {
+    table_id: Hex;
+    field_layout: Hex;
+    key_schema: Hex;
+    value_schema: Hex;
+    abi_encoded_key_names: Hex;
+    abi_encoded_field_names: Hex;
+  }[];
+};
+
+export function useDeployedTablesQuery() {
+  const { worldAddress } = useParams();
+  const TABLES_QUERY_TABLE = `${worldAddress}__store__tables`;
+
+  return useQuery({
+    queryKey: ["deployedTables", TABLES_QUERY_TABLE],
+    queryFn: async () => {
+      const response = await fetch(`/api/table?tableId=${TABLES_QUERY_TABLE}`);
```

Comment:
what if this endpoint returned a well-formed `Table` object instead of requiring the frontend to compute this? then the frontend only has to know about `Table`

Reply:
yeah, this might not the right level to do it at, and could be moved up to the endpoint. However, I'm also thinking that in case the local indexer exposes the same endpoint that matches Dozer one, it would be convenient to use the same query, just with different endpoints. Looking more into this